src/main.rs

```diff
@@ -73,15 +74,44 @@ fn remove(opt: cli::Opt, key: String) -> Result<()> {
     config.save(&opt.amber_yaml)
 }
 
-fn print(opt: cli::Opt) -> Result<()> {
+fn print(opt: cli::Opt, style: Option<cli::PrintStyle>) -> Result<()> {
     let config = config::Config::load(&opt.amber_yaml)?;
     let secret = config.load_secret_key()?;
     let pairs: Result<Vec<_>> = config.iter_secrets(&secret).collect();
     let mut pairs = pairs?;
     pairs.sort_by(|x, y| x.0.cmp(y.0));
-    pairs
-        .iter()
-        .for_each(|(key, value)| println!("export {}={:?}", key, value));
+
+    if let Some(style) = style {
+        let to_obj = |p: Vec<(&String, String)>| {
```

Comment:
I'm having trouble parsing this code, or at least understanding what it's trying to achieve. Why is this creating a `Vec` of `HashMap`s?

Reply:
Maybe it is better to be `to_objs`?

The logic of this part is converting `[ (KEY, value) ]` to `[ { object.key = "KEY", object.value = "value" } ]`.

Reply:
Oh, got it, thanks. A comment explaining _why_ this is the desired output would be good. I'm assuming it's to keep correct sorting of the data?